src/rest.js

```diff
@@ -42,6 +42,18 @@ function del(config, auth, className, objectId) {
 
   enforceRoleSecurity('delete', className, auth);
 
+  if (objectId === '*') {
+    return Promise.resolve().then(() => {
+      return config.database.purgeCollection(className);
+    }).then(() => {
+      if (className == '_Session') {
+        var cacheAdapter = config.cacheController;
+        cacheAdapter.user.clear();
```

Comment:
We'll need to do something similar for role cache. Probably also worth adding a test to make sure that deleting all objects actually causes the cache to get cleared (eg. create a role, add a user to it, delete all roles using this endpoint, and ensure that the user no longer has access to objects that only that role could access)